src/libcore/iter.rs

```diff
@@ -280,6 +289,16 @@ pub trait Iterator {
     /// different sized integer, the `zip` function provides similar
     /// functionality.
     ///
+    /// # Undefined overflow
+    ///
+    /// The method does no guarding against overflows, so enumerating more than
+    /// `usize::MAX` elements is undefined.
```

Comment:
[The RFC](https://github.com/rust-lang/rfcs/blob/master/text/0560-integer-overflow.md#arithmetic-operations-with-error-conditions) and I believe the current implementation both define overflow in the case of optimized builds to be the normal 2's complement overflow.